zerver/actions/streams.py

```diff
@@ -1686,11 +1698,14 @@ def do_change_stream_group_based_setting(
     if setting_name in Stream.stream_permission_group_settings_granting_metadata_access:
         old_user_ids_with_metadata_access = can_access_stream_metadata_user_ids(stream)
 
-    if isinstance(new_setting_value, NamedUserGroup):
-        user_group: UserGroup = new_setting_value
+    if isinstance(new_setting_value, int):
```

Comment:
Can we avoid doing this here and instead call `access_user_group_api_value_for_setting` before calling `do_change_stream_group_based_setting`?